src/binheap.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding utf-8 -*-
+
+
+class Binheap(object):
+
+    def __init__(self, iterable=None):
+        '''Initialise a heap from an iterable or an empty heap'''
+        if iterable is None:
+            self.heap = None
+        else:
+            self.heap = sorted(iterable)
+
+    def push(self, val):
+        '''puts new value ito the heap, maintaining the heap property'''
+        if val is None:
```

Comment:
This doesn't quite match up. Your error is a ValueError telling the user that it requires a numerical value only (e.g. int or float), but only checking if the pushed value is None. So if the user pushed a string, it would not be an error. Is this what you want?

Secondly, if you want to raise this kind of error, it should be a TypeError (input data type is wrong)